src/error.rs

```diff
@@ -80,15 +80,18 @@ impl CompileError {
     pub fn add_additional_message(
         mut self,
         file: &SimpleFile<String, String>,
-        range: Range<usize>,
+        range: Option<Range<usize>>,
```

Comment:
I'm currently in process on a commit to try to get rid of passing `Option<Range>`s  into errors.  It's certainly non-trivial, but the idea in my in progress commit is to force the caller to handle the None case in a more sane way, since the error layer doesn't have the context to understand what it means when the Range is none.

I haven't really looked much at the callers, but is the None case something that can be handled at the call site?  Perhaps by pointing at the thing that made a synthetic type with a None Range?